Add LAPACK++

ax3l opened the pull request:
Add LAPACK++: modern C++ bindings for LAPACK libraries.

Applied patches are proposed to upstream in:
* https://bitbucket.org/icl/lapackpp/pull-requests/13/msvc-windows-builds

Follow-up to #14176

<!--
Thank you very much for putting in this recipe PR!

This repository is very active, so if you need help with
a PR or once it's ready for review, please let the right people know.
There are language-specific teams for reviewing recipes.

Currently available teams are:
- python `@conda-forge/help-python`
- python/c hybrid `@conda-forge/help-python-c`
- r `@conda-forge/help-r`
- java `@conda-forge/help-java`
- nodejs `@conda-forge/help-nodejs`
- c/c++ `@conda-forge/help-c-cpp`
- perl `@conda-forge/help-perl`
- Julia `@conda-forge/help-julia`
- ruby `@conda-forge/help-ruby`

If your PR doesn't fall into those categories please contact
the full review team `@conda-forge/staged-recipes`.

Due to GitHub limitations first time contributors to conda-forge are unable
to ping these teams. You can [ping the team][1] using a special command in
a comment on the PR to get the attention of the `staged-recipes` team. You can
also consider asking on our [Gitter channel][2] if your recipe isn't reviewed promptly.

[1]: https://conda-forge.org/docs/maintainer/infrastructure.html#conda-forge-admin-please-ping-team
[2]: https://gitter.im/conda-forge/conda-forge.github.io
-->

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] ~If static libraries are linked in, the license of the static library is packaged.~
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

cc @G-Ragghianti @mgates3

I am adding this for our WarpX desktop users as mentioned per e-mail. No need to get active from your side, but if anyone of you wants to co-maintain this at any point in the future do not hesitate to reach out and we'll add you.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/lapackpp```) and found it was in an excellent condition.

ax3l commented:
The CMake install path needs a slight update, added to upstream:
- https://bitbucket.org/icl/blaspp/pull-requests/31/msvc-windows-builds
and fixed in:
- https://github.com/conda-forge/blaspp-feedstock/pull/1

ax3l commented:
On, and library symbols are hidden on Windows by  default, this fixes it (also added to above linked upstream PR):* 
* https://github.com/conda-forge/blaspp-feedstock/pull/2

ax3l commented:
@conda-forge/help-c-cpp ready for review/merge :)
